src/main/java/com/microtripit/mandrillapp/lutung/MandrillApi.java

```diff
@@ -37,36 +26,30 @@
 	private final MandrillExportsApi exports;
 	private final MandrillIpsApi ips;
 	
-	public MandrillApi(final String key) {
+	public MandrillApi(final String key, final String rootUrl) {
 		if(key == null) {
 			throw new NullPointerException(
 					"'key' is null; please provide Mandrill API key");
 			
 		}
-		this.key = key;
-		users = new MandrillUsersApi(key);
-		messages = new MandrillMessagesApi(key);
-		tags = new MandrillTagsApi(key);
-		rejects = new MandrillRejectsApi(key);
-		whitelists = new MandrillWhitelistsApi(key);
-		senders = new MandrillSendersApi(key);
-		urls = new MandrillUrlsApi(key);
-		templates = new MandrillTemplatesApi(key);
-		webhooks = new MandrillWebhooksApi(key);
-		subaccounts = new MandrillSubaccountsApi(key);
-		inbound = new MandrillInboundApi(key);
-		exports = new MandrillExportsApi(key);
-		ips = new MandrillIpsApi(key);
+		final MandrillService mandrillService = new MandrillService(rootUrl, new MandrillRequestDispatcher(), key);
```

Comment:
to preserve the logic from the above check, maybe we can throw a NPE or invalidsomething when the rootUril is null in the constructor of the service?